src/individual/CalculateBallShakes.c

```diff
@@ -47,7 +48,11 @@ u16 MoonBallSpecies[] =
  *  @return the amount of shakes a ball undergoes.  or'd with 0x80 for critical captures
  */
 u32 CalculateBallShakes(void *bw, struct BattleStruct *sp) {
-    u32 i, captureRate, ballRate, type1, type2, criticalCapture = FALSE;
+    u32 i, speciesCatchRate, ballCaptureRatio, type1, type2, criticalCapture = FALSE;
+    u32 heavyBallMod = 0;
+    u32 a = 0, b = 0, c = 0, d = 0, e = 0, f = 0, g = 0, captureValueCoeffcient = 0, modifiedCatchRate = 0, badgePenalty = 4096, statusModifier = 0, criticalCatchModifier = 0, speciesInDex = 0, criticalCatchRate = 0, shakeChecks = 4, shakeChance = 0;
```

Comment:
should badge penalty be a q412 constant?  applies to all `ballCaptureRatio` below as well

Reply:
Yes